src/Catalog.API/Extensions/Extensions.cs

```diff
@@ -27,12 +28,15 @@ public static void AddApplicationServices(this IHostApplicationBuilder builder)
         builder.Services.AddOptions<CatalogOptions>()
             .BindConfiguration(nameof(CatalogOptions));
 
-        builder.Services.AddOptions<AIOptions>()
-            .BindConfiguration("AI");
-
-        if (!string.IsNullOrWhiteSpace(builder.Configuration.GetConnectionString("openai")))
+        if (builder.Configuration["AI:Onnx:EmbeddingModelPath"] is string modelPath &&
+            builder.Configuration["AI:Onnx:EmbeddingVocabPath"] is string vocabPath)
+        {
+            builder.Services.AddBertOnnxTextEmbeddingGeneration(modelPath, vocabPath);
+        }
+        else if (!string.IsNullOrWhiteSpace(builder.Configuration.GetConnectionString("openai")))
         {
             builder.AddAzureOpenAIClient("openai");
+            builder.Services.AddOpenAITextEmbeddingGeneration(builder.Configuration["AIOptions:OpenAI:EmbeddingName"] ?? "text-embedding-3-small");
```

Comment:
Can you also change this in the AppHost so they are aligned?

https://github.com/dotnet/eShop/blob/d052bab01059e7e72bd0736fc63336d12a981551/src/eShop.AppHost/Program.cs#L77-L100